python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -134,16 +134,15 @@ cpdef generate_pandas_metadata(Table table, index):
                 index_descriptors.append(descr)
             else:
                 col_names.append(name)
-
+    metadata_df = table.head(0).to_pandas()
```

Comment:
Yes, I felt like this was a pretty bad hack. The pyarrow function here writes the metadata by subbing in the names of the dtype objects attached to the dataframe. So if we pass it our frame it writes a parquet file that other readers can't read because they don't understand the dtypes.